puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/testKatello.groovy

```diff
@@ -0,0 +1,141 @@
+                   doGenerateSubmoduleConfigurations: false,
+                   extensions: [[$class: 'PreBuildMerge', options: [fastForwardMode: 'FF', mergeRemote: 'origin', mergeStrategy: 'default', mergeTarget: '${ghprbTargetBranch}']]],
+                   userRemoteConfigs: [
+                       [refspec: '+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*', url: 'https://github.com/katello/katello']
+                   ]
+               ]
+
+                dir('foreman') {
+                    git url: "https://github.com/theforeman/foreman", branch: katello_versions[ghprbTargetBranch]
+                }
+            }
+        }
+        stage("Setup RVM") {
+            steps {
+
+                configureRVM(ruby)
+
+            }
+        }
+        stage('Configure Environment') {
+            steps {
+
+                dir('foreman') {
+                    addSettings([
+                        organizations: true,
+                        locations: true
+                    ])
+                    addGem()
+                    databaseFile(gemset())
+                }
+
+            }
+        }
+        stage('Configure Database') {
+            steps {
+
+                dir('foreman') {
+                    configureDatabase(ruby)
+                }
+
+            }
+        }
+        stage('Run Tests') {
+            parallel {
+                stage('tests') {
+                    steps {
+                        dir('foreman') {
+                            withRVM(['bundle exec rake jenkins:katello TESTOPTS="-v"'], ruby)
+                        }
+                    }
+                }
+                stage('rubocop') {
+                    steps {
+                        dir('foreman') {
+                            withRVM(['bundle exec rake katello:rubocop TESTOPTS="-v" --trace'], ruby)
+                        }
+                    }
+                }
+                stage('react-ui') {
+                    when {
+                        expression { fileExists('package.json') }
+                    }
+                    steps {
+                        sh 'npm install'
+                        sh 'npm test'
+                    }
+                }
+                stage('angular-ui') {
+                    steps {
+                        dir('foreman') {
+                            withRVM(['bundle show bastion > bastion-version'], ruby)
+
+                            script {
+                                bastion_install = readFile('bastion-version')
+                                bastion_version = bastion_install.split('bastion-')[1]
+                                echo bastion_install
+                                echo bastion_version
+                            }
+                        }
+
+                        sh "cp -rf \$(cat foreman/bastion-version) engines/bastion_katello/bastion-${bastion_version}"
```

Comment:
can this be `sh "cp -rf ${bastion_install} engines/bastion_katello/bastion-${bastion_version}"`?

Reply:
or even `sh "cp -rf ${bastion_install} engines/bastion_katello/${bastion_install}"`?

Reply:
I wanted it to be. But for some reason it seems like the fact that there are special characters in the path such as `@` throws off this variable substitution causing groovy to not create the right string :/ I tried 4-5 different ways to avoid this intermediate file.

Reply:
Uh, well, that explains it. Then let's go with the ugly :cat: